packages/manager/modules/pci/src/projects/project/kubernetes/details/service/oidc-provider/add-oidc-provider/routing.js

```diff
@@ -0,0 +1,32 @@
+export default /* @ngInject */ ($stateProvider) => {
+  $stateProvider.state(
+    'pci.projects.project.kubernetes.details.service.add-oidc-provider',
+    {
+      url: '/add-oidc-provider',
+      views: {
+        modal: {
+          component: 'pciProjectKubernetesServiceAddOidcProvider',
+        },
+      },
+      layout: 'modal',
+      redirectTo: (transition) =>
+        transition
+          .injector()
+          .getAsync('oidcProvider')
+          .then((oidcProvider) => {
+            return oidcProvider.isDefined()
+              ? { state: 'pci.projects.project.kubernetes.details.service' }
+              : false;
+          }),
+      resolve: {
+        breadcrumb: () => null,
+
+        oidcProviderModel: /* @ngInject */ (oidcProvider) =>
```

Comment:
Why is a copy needed ?

Reply:
To avoid to update the current provider.